Hero battle Michalis

AlmostMatt opened the pull request:
Adds an additional set of heroes (manually entered) to stats.json as grand_battle_heroes, and shows them at the start of the hero list in the app

AlmostMatt committed:
Add 113 and 150px icons for AI enemies

AlmostMatt committed:
Add Michalis hero battle lunatic stats to stats.json and add support for imageName not equal to hero name

AlmostMatt committed:
Separate herogrid (row) for grand-battle heroes

ajhyndman commented on packages/fire-emblem-heroes-stats/lib/stats.json:
```diff
@@ -1,4 +1,243 @@
+  "grand_battle_heroes": [
+    {
+      "name": "Flier",
+      "imageName": "Lance Flier",
```
I'm thinking that I like the idea of a `shortName` rather than `imageName` on the hero.

The semantics seem slightly more correct.  The only reason you're diverging the two keys is so that you can consume the abbreviated version in spaces in the UI where a longer name wouldn't fit.  It's a bit weird if the only full, canonical name is in the `imageName` key.

ajhyndman reviewed (commented):
Two further changes I'm going to try to make:

## Flow Type
We lose a lot of the benefits of flow if the data gets out of sync with the type definition, so I'll update that too.

## Relocate Hero Battle stats
I think it's nice that this module's stats are a pure function of the script.  I'd like to be able to think of the `/lib` folder as a compiled output.  I propose either:
* put these stats in`fire-emblem-stats/src/` and updating the script to compile them into this file, or
* put the stats directly in `proving-grounds` and consume them from there.

ajhyndman commented:
All the images look great!

ajhyndman commented:
Actually, I think I might be misusing some of the standard node package structure conventions.  I think my `/lib` should probably be `/dist`.  `/src` should probably be `/build`.  It's not a big deal, but theoretically makes the project more accessible for potential contributors.

https://gist.github.com/tracker1/59f2c13044315f88bee9

ajhyndman committed:
Relocate michaelis unit stats, and update types

ajhyndman committed:
Fix stat shape errors

This'll be important for compatibility with hero configuration.

Also run eslint --fix

ajhyndman committed:
imageName -> shortName

ajhyndman commented:
There are two flow errors remaining, but fixing them is going to slightly impede merge resolution with my rarity gui, so I'm just going to leave alone for now.

```
E:\Projects\fire-emblem-working-title\packages\proving-grounds [hero-battle ≡]
[01:55] λ node_modules\.bin\flow
src/heroHelpers.js:60
 60:     const value = parseInt(hero.stats[level][rarity][statKey], 10);
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ access of computed property/element. Computed property/element cannot be accessed on possibly undefined value
 60:     const value = parseInt(hero.stats[level][rarity][statKey], 10);
                                ^^^^^^^^^^^^^^^^^^^^^^^^^ undefined

src/heroHelpers.js:67
 67:   const values = hero.stats[level][rarity][statKey];
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ access of computed property/element. Computed property/element cannot be accessed on possibly undefined value
 67:   const values = hero.stats[level][rarity][statKey];
                      ^^^^^^^^^^^^^^^^^^^^^^^^^ undefined


Found 2 errors
```

ajhyndman commented:
@AlmostMatt thanks for setting this up as a PR and giving me a chance to comment.  👍 

What do you think of my changes?  I've tried to break them into chunks as commits so we can pick and choose.
